### ISSUE TITLE
(Flaky-test) AuthPluginTest.testAthenz failure

### ISSUE THREAD

**Issue by devinbost** [apache/pulsar#6298]
This is a new exception. (I haven't seen this test fail before.)

Here's what appeared in the logs:
```
[147/161] AuthPluginTest.testAthenz (315 ms)
174Note: Google Test filter = AuthPluginTest.testAthenz
175[==========] Running 1 test from 1 test case.
176[----------] Global test environment set-up.
177[----------] 1 test from AuthPluginTest
178[ RUN ] AuthPluginTest.testAthenz
1792020-02-12 00:18:09.900 INFO AuthPluginTest:245 | PARAMS: {
180 "tenantDomain": "pulsar.test.tenant",
181 "tenantService": "service",
182 "providerDomain": "pulsar.test.provider",
183 "privateKey": "file:../../pulsar-broker/src/test/resources/authentication/tls/client-key.pem",
184 "ztsUrl": "http://localhost:9999"
185 }
1862020-02-12 00:18:09.907 ERROR ZTSClient:350 | Response failed for url http://localhost:9999/zts/v1/domain/pulsar.test.provider/token. Error Code 7
187/pulsar/pulsar-client-cpp/tests/AuthPluginTest.cc:251: Failure
188Value of: "Athenz-Role-Auth: mockToken"
189Expected: data->getHttpHeaders()
190Which is: "Athenz-Role-Auth: "
191terminate called without an active exception
```

It can be seen here: 
https://github.com/apache/pulsar/pull/6202/checks?check_run_id=439900732

I've attached the logs.
[logs_16021.zip](https://github.com/apache/pulsar/files/4189300/logs_16021.zip)

**Comment by merlimat** [comment]
I'm seeing this rather frequently. @massakam Can you take a look?

**Comment by lhotari** [comment]
A recent failure: https://github.com/apache/pulsar/runs/6723820880?check_suite_focus=true#step:11:52
```
Note: Google Test filter = AuthPluginTest.testAthenz
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from AuthPluginTest
[ RUN      ] AuthPluginTest.testAthenz
2022-06-03 10:14:54.269 INFO  [140447679827648] AuthPluginTest:240 | PARAMS: {
        "tenantDomain": "pulsar.test.tenant",
        "tenantService": "service",
        "providerDomain": "pulsar.test.provider",
        "privateKey": "file:../../pulsar-broker/src/test/resources/authentication/tls/client-key.pem",
        "ztsUrl": "http://localhost:9999"
    }
2022-06-03 10:14:54.286 ERROR [140447679827648] ZTSClient:367 | Response failed for url http://localhost:9999/zts/v1/domain/pulsar.test.provider/token. Error Code 7
/pulsar/pulsar-client-cpp/tests/AuthPluginTest.cc:246: Failure
Expected equality of these values:
  data->getHttpHeaders()
    Which is: "Athenz-Role-Auth: "
  "Athenz-Role-Auth: mockToken"
terminate called without an active exception
[6/295] AuthPluginTest.testAthenz returned/aborted with exit code -6 (567 ms)
[7/295] AuthPluginTest.testAuthFactoryTls (542 ms)
[8/295] AuthPluginTest.testAuthFactoryAthenz (447 ms)
[9/295] AuthPluginTest.testAthenz (355 ms)
Note: Google Test filter = AuthPluginTest.testAthenz
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from AuthPluginTest
[ RUN      ] AuthPluginTest.testAthenz
2022-06-03 10:14:54.802 INFO  [140642816[53](https://github.com/apache/pulsar/runs/6723820880?check_suite_focus=true#step:11:54)6256] AuthPluginTest:240 | PARAMS: {
        "tenantDomain": "pulsar.test.tenant",
        "tenantService": "service",
        "providerDomain": "pulsar.test.provider",
        "privateKey": "file:../../pulsar-broker/src/test/resources/authentication/tls/client-key.pem",
        "ztsUrl": "http://localhost:9999"
    }
2022-06-03 10:14:[54](https://github.com/apache/pulsar/runs/6723820880?check_suite_focus=true#step:11:55).810 ERROR [140642816536256] ZTSClient:367 | Response failed for url http://localhost:9999/zts/v1/domain/pulsar.test.provider/token. Error Code 7
/pulsar/pulsar-client-cpp/tests/AuthPluginTest.cc:246: Failure
Expected equality of these values:
  data->getHttpHeaders()
    Which is: "Athenz-Role-Auth: "
  "Athenz-Role-Auth: mockToken"
terminate called without an active exception
[9/295] AuthPluginTest.testAthenz returned/aborted with exit code -6 (3[55](https://github.com/apache/pulsar/runs/6723820880?check_suite_focus=true#step:11:56) ms)
```

**Comment by lhotari** [comment]
I pushed a fix in #15924
